src/_gettsim/transfers/rente.py

```diff
@@ -770,15 +916,17 @@ def ges_rente_vorauss_langj(
     Eligibility as bool.
 
     """
-    out = (alter >= ges_rente_params["altersgrenze_langj_versicherte_vorzeitig"]) and (
-        ges_rente_wartezeit_35 >= 35
-    )
+    out = (
+        alter >= ges_rente_params["altersgrenze_langj_versicherte_vorzeitig"]
+    ) and ges_rente_wartezeit_35
 
     return out
 
 
 @dates_active(start="2012-01-01")
-def ges_rente_vorauss_besond_langj(ges_rente_wartezeit_45: float) -> bool:
+def ges_rente_vorauss_besond_langj(
```

Comment:
Just curious why this is structured differently from `ges_rente_vorauss_langj` ? I.e., why is the age check done elsewhere and this function does not do anything really?

Reply:
yeah, this had to do with the if condition in the zugangsfaktor function.  The ```ges_rente_vorauss_vorzeitig```.  But since I stumbled across this many times and wasted quite some time by forgetting and remind myself. Also we could not test for early retirement thresholds for non retired people that way, without making very strange test cases. 

 I resolved this with commit [93be5f8](https://github.com/iza-institute-of-labor-economics/gettsim/pull/634/commits/93be5f8c1c7004fd20b6e506fa85b7bcd13034ba).

 The age check is now done separately now. There I run into rounding issues of the retirement age that interferred with the age check. It works now with the existing test cased but. Might need to visit this again.

Reply:
outdated